allennlp/data/token_indexers/pretrained_transformer_mismatched_indexer.py

```diff
@@ -64,6 +64,9 @@ def tokens_to_indices(self, tokens: List[Token], vocabulary: Vocabulary) -> Inde
         output = self._matched_indexer.tokens_to_indices(tokens, vocabulary)
 
         # self._intra_word_tokenize() does not insert special tokens, so we need to do it here
```

Comment:
This comment refers to the line below your diff. Could you move it below?

Reply:
Both of those lines refer ti the insertion of special tokens. Those functions are unfortunately named, but they do the same thing. One does it for token ids, and the other does it for type ids.

Reply:
I'm pretty confused by this... The docs seem to say that this does exactly what it sounds like, create type ids. See https://huggingface.co/transformers/model_doc/bert.html#transformers.BertTokenizer.create_token_type_ids_from_sequences What special tokens are inserted?

Reply:
Okay, I ran the following and see what you mean, but the comment is still really confusing/broken now.

```
In [1]: import transformers

In [2]: tokenizer = transformers.tokenization_auto.AutoTokenizer.from_pretrained("bert-base-cased")

In [3]: tokenizer.build_inputs_with_special_tokens([1,2,3,4])
Out[3]: [101, 1, 2, 3, 4, 102]

In [4]: tokenizer.create_token_type_ids_from_sequences([1,2,3,4])
Out[4]: [0, 0, 0, 0, 0, 0]
```

How about:

```
self._intra_word_tokenize() does not insert special tokens, so we need to do it here. Before this we need to create the type ids as create_token_type_ids_from_sequences expects no special tokens -- though it does insert their corresponding ids.
```

Reply:
I added two comments. f04588327419209610795288815b49d983d7bf06